includes/Jobs/GenerateProductFeed.php

```diff
@@ -56,13 +56,46 @@ protected function handle_start() {
 		$this->feed_file_handler->write_to_temp_file(
 			$this->feed_data_exporter->generate_header()
 		);
+		// Reset the statistics counters.
+		facebook_for_woocommerce()->get_tracker()->track_feed_file_batch_generation_average_time( 0 );
+		facebook_for_woocommerce()->get_tracker()->track_feed_file_batch_count( 0 );
+		facebook_for_woocommerce()->get_tracker()->track_feed_file_generation_start( time() );
+	}
+
+	/**
+	 * Handle processing a chain batch.
+	 *
+	 * @hooked {plugin_name}/jobs/{job_name}/chain_batch
+	 *
+	 * @param int   $batch_number The batch number for the new batch.
+	 * @param array $args         The args for the job.
+	 *
+	 * @throws Exception On error. The failure will be logged by Action Scheduler and the job chain will stop.
+	 */
+	public function handle_batch_action( int $batch_number, array $args ) {
+		$start_time = microtime( true );
+		parent::handle_batch_action( $batch_number, $args );
+		$generation_time       = microtime( true ) - $start_time;
+		$previous_average_time = facebook_for_woocommerce()->get_tracker()->get_feed_batch_generation_average_time();
```

Comment:
I think it's best to avoid updating transients that will be used for tracking in the middle of the batch. There is a risk that the snapshot will contain partial data.

I'd suggest `GenerateProductFeed` uses its own (private) transient to accumulate the data, and then update the tracker on completion.